src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescriptionJsonConverter.cs

```diff
@@ -14,6 +15,8 @@ namespace Microsoft.Identity.Abstractions
     /// </summary>
     /// <remarks>BE CAREFUL when you serialize a credential description. The secrets it contains
     /// will be serialized too depending on the CredentialSource (ClientSecret, Base64Encoded, and Password)</remarks>
+    [RequiresDynamicCode("Uses JsonSerializer which may require dynamic code generation for certain types.")]
+    [RequiresUnreferencedCode("Uses JsonSerializer which may require dynamic code generation for certain types.")]
```

Comment:
For DecryptKeysAuthenticationOptions, we could check if the options has a TypeInfoResolver, and if it doesn't then we could clone the options, and create a source generated context.